Form/Type/AdminType.php

```diff
@@ -67,11 +67,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                             $subject = $subjectCollection->get(trim($options['property_path'], '[]'));
                         }
                     } else {
+                        // this check is to work around duplication issue in property path
+                        // https://github.com/sonata-project/SonataAdminBundle/issues/4425
+                        if ($this->getFieldDescription($options)->getFieldName() === $options['property_path']) {
+                            $path = $options['property_path'];
+                        } else {
+                            $path = $this->getFieldDescription($options)->getFieldName().$options['property_path'];
```

Comment:
you're right - this is actually about embedded admins

-> the other issue was about duplicating the property path, this is why I kept the check for this. But at the same time it broke the subject finding in the embedded admin